### PR TITLE
feat(eslint-plugin-wantedly): Add new rule 'no-data-testid'

### PR DESCRIPTION
**Pull request by chloe463**
## Why

`data-testid` is useful to test components.
However, if it remains on our production code, it is going to be easier for those who use RPA or some kind of automation tools to crawl/scrape the page.
So we use [@welldone-software/babel-plugin-react-add-test-id](https://www.npmjs.com/package/@welldone-software/babel-plugin-react-add-test-id) to add 'data-testid' to each component on test environment only.
But still there are possibilities that the attribute sneak in our code.
Maybe we can reject those codes by reviewing but it is hard to do that with a 100% possibility.
So I want to introduce this rule to reject `data-testid`

## What

Add new rule `no-data-testid`.

valid
```
<div />
```

invalid
```
<div data-testid="foo" />
```

### CODE REVIEW

**Comment by chloe463** [comment]
## TODO

- [ ] Make type definition better
- [ ] Write doc

**Comment by chloe463** [comment]
@yamadayuki 
What do you think about this?
I searched for an eslint rule plugin like this, but I could not find one, so I made it.
Before I proceed to do the tasks above I want to know your opinion.
If you know something similar to this one or have any good ideas, please let me know.

**Comment by yamadayuki** [comment]
I agreed with your suggestion. I don't know something like to forbid using some attribute such as `data-testid`. Plz go ahead.

**Comment by chloe463** [comment]
I'm glad to hear that.
I'll complete this PR in couple of days.
Thanks 👍

**Comment by chloe463** [comment]
@yamadayuki 
Could you review this?

File: packages/eslint-plugin-wantedly/src/rules/no-data-testid.ts [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+            context.report({
+              loc: node.loc!,
+              message: "Attribute {{key}} is not allowed.",
+              data: {
+                key,
+              },
+            });
+          }
+        });
+        return;
```
Comment (yamadayuki) [inline packages/eslint-plugin-wantedly/src/rules/no-data-testid.ts:39]: Redundant return statement.

File: packages/eslint-plugin-wantedly/src/rules/no-data-testid.ts [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+export const RULE_NAME = "no-data-testid";
+
+const DEFAULT_OPTION = {
+  denyKeyList: ["data-testid"],
+};
+
+linter.defineRule(RULE_NAME, {
+  meta: {
+    type: "suggestion",
+    fixable: "code",
```
Comment (yamadayuki) [inline packages/eslint-plugin-wantedly/src/rules/no-data-testid.ts:16]: This rule is not fixable. Omit this line.

File: packages/eslint-plugin-wantedly/docs/rules/no-data-testid.md [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+## Options
+
+#### `denyKeyList`
+
+The list of name which is not allowed to assign as prop. Default is `["data-testid"]`.
+
+```json
+{
+  "rule": {
+    "wantedly/no-data-testid": ["error": { "denyKeyList": ["data-testid", "data-test-id"] }]
```
Comment (yamadayuki) [inline packages/eslint-plugin-wantedly/docs/rules/no-data-testid.md:38]: I think the `denyKeyList` is not only aim to deny the `data-testid`. If the user specify the `key` in this rule option, the user prohibit to set the key in the JSX Element. So I don't know the rule name `wantedly/no-data-testid` is appropriate.

File: packages/eslint-plugin-wantedly/src/rules/no-data-testid.ts [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+  // cf. https://github.com/DefinitelyTyped/DefinitelyTyped/blob/a0b66498155af79ddc7cdb52dba529eb5b04eae8/types/eslint/index.d.ts#L241-L266
+  /* @ts-ignore */
+  create(context) {
+    const option = getOptionWithDefault(context, DEFAULT_OPTION);
+    return {
+      JSXOpeningElement: (node: JSXOpeningElement) => {
+        option.denyKeyList.forEach((key: string) => {
+          if (hasProp(node.attributes, key)) {
+            context.report({
+              loc: node.loc!,
```
[the remark this answers is not in the document]
Comment (yamadayuki) [reply packages/eslint-plugin-wantedly/src/rules/no-data-testid.ts:31]: And `node.loc` is the `JSXOpeningElement` node's location but not the denied attribute location. We should tell the user to fix the lint issue at the attribute location.

**Pull request closed without merging**